### ISSUE TITLE
Import gifs as costumes

### ISSUE THREAD

**Issue by 223SK** [jmoenig/Snap--Build-Your-Own-Blocks#958]
Scratch lets you import a gif and add each frame as a costume. This allows for much more efficiency in animation projects. Currently, SNAP only imports the first frame.

**Comment by jmoenig** [comment]
Yep, that's true. But you *can* import more than one picture at once. I always create a folder for all frames of my stop motion pictures (usually from the camera), select them all and drag them over into Snap all at once. Please let me know if that doesn't work for you. Thanks!

**Comment by brianharvey** [comment]
Hmm, are they guaranteed to be imported in the right order?

**Comment by jmoenig** [comment]
The sorting is handled by the OS. I'm assuming that it's the order in which you mark them, but I don't know, since it actually *is* something Snap gets directly from the OS...

**Comment by brianharvey** [comment]
Then it's probably alphabetical order.  Try an example with 10 frames and see if you get foo1, foo10, foo2, foo3, etc.

**Comment by jmoenig** [comment]
You know, you can also manually sort costumes in the costumes tab by dragging and rearranging them (e.g. in the case of foo1, foo10, foo2)

**Comment by cycomachead** [comment]
> I always create a folder for all frames of my stop motion pictures (usually from the camera), select them all and drag them over into Snap all at once. Please let me know if that doesn't work for you.

This doesn't work for gifs because it's not terribly easy to split a gif.

**Comment by bromagosa** [comment]
We've looked into this for BeetleBlocks, because we wanted to use animated GIFs in the UI for tutorials.

It is currently impossible to access individual frames in a GIF from in HTML5. The approach people are using is to send the file to a server that processes it into separate images and sends them back, which is not an option here.

**Comment by cycomachead** [comment]
> It is currently impossible to access individual frames in a GIF in HTML5

Why can't you parse a gif file with `readAsBinaryData()`

I have seen hacks which play a GIF in a hidden canvas and capture different frames, but that's very much an annoying work around.

Still, this would be terribly useful for many people, and since it's not a necessary feature, I don't see why we couldn't use an external service, or at least recommend one... Splitting a GIF on the computer isn't a terribly easy process.
